docs/worker-yaml.md

```
@@ -0,0 +1,226 @@
+|----|----|
+| Env variable unset and no default | Startup fails (pre-validation) |
+| `$file:` target unreadable | Startup fails (pre-validation) |
+| `$file:` resolves outside `base_dir` | Startup fails (pre-validation) |
+| Top-level / worker / retry_policy field typo | Startup fails (serde parse stage) |
+| Invalid enum string | Startup fails (pre-validation) |
+| Duplicate `name` within one YAML | Startup fails (pre-validation) |
+| Runner name not registered on jobworkerp | Startup fails (pre-validation) |
+| `settings` does not match the runner schema | Startup fails (pre-validation) |
+| Worker has `settings` but the runner declares no `runner_settings` schema | Startup fails (pre-validation) |
+| `settings` contains a key that is not in the runner's `runner_settings` proto schema (typo) | Startup fails (pre-validation — `ignore_unknown_fields=false` for declarative registration) |
+| Env value violates the structure-safety guard (line break, tab, `": "`, `" #"`, `,`, `]`, `}` anywhere; or starts with `#`, `&`, `*`, `!`, `|`, `>`, `[`, `{`, `'`, `"`, `` ` ``, `\`, `?`, `"- "`, `"% "`; or matches a bool/null literal) | Startup fails (pre-validation — see "Structure-safety guard") |
```

markdownlint-cli2 (0.22.1)

[warning] Spaces inside code span elements

(MD038, no-space-in-code)